examples/Workflow/WorkflowWebApp/Program.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Dapr.Workflow;
+
+// The workflow host is a background service that connects to the sidecar over gRPC
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+
+// Dapr workflows are registered as part of the service configuration
+builder.Services.AddWorkflow(options =>
+{
+    // Example of registering a "Hello World" workflow function
+    options.RegisterWorkflow<string, string>("HelloWorld", implementation: async (context, input) => 
+    {
+        return await context.CallActivityAsync<string>("SayHello", "World");
+    });
+
+    // Example of registering a "Say Hello" workflow activity function
+    options.RegisterActivity<string, string>("SayHello", implementation: (context, input) =>
+    {
+        return Task.FromResult($"Hello, {input}!");
+    });
+});
+
+WebApplication app = builder.Build();
+
+// POST starts new workflow instances
+app.MapPost("/workflow", async (HttpContext context, WorkflowClient client) =>
+{
+    string id = Guid.NewGuid().ToString()[..8];
+    await client.ScheduleNewWorkflowAsync("HelloWorld", id);
+
+    // return an HTTP 202 and a Location header to be used for status query
+    return Results.AcceptedAtRoute("GetWorkflowEndpoint", new { id });
+});
+
+// GET fetches metadata for specific workflow instances
+app.MapGet("/workflow/{id}", async (string id, WorkflowClient client) =>
```

Comment:
I think that this example, as a web application might be making things a bit more complicated than they need to be.

It does make it simpler to handle the async nature of workflow execution by delegating the "status checking" to the user running the example. I think this should rather be a CLI that keeps pooling for a status change (just to make it more... interactive 😅), prints things as it goes and is done with it.

Thoughts?

Reply:
This example is confusing because it treats workflows as first-class things, whereas a real app doesn't think in terms of workflows, but rather in terms of "orders", "jobs", etc. Ideally, this route would be something like "/orders/{orderId}" where the implementation maps the orderId to a workflow ID.

Reply:
I did a small renaming inside this file which should be sent along the next commit.